Task2/classification.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import numpy as np
+import pandas as pd
+from sklearn import preprocessing
+from sklearn import tree, linear_model, svm, naive_bayes
+from sklearn.metrics import precision_recall_fscore_support, average_precision_score, accuracy_score
+from sklearn.model_selection import GridSearchCV
+
+import warnings
+warnings.filterwarnings('ignore')  # "error", "ignore", "always", "default", "module" or "once"
+
+titles_final = ['Mr', 'Mrs', 'Ms', 'Master', 'Col', 'Sir', 'Lady', 'Dr', 'Rev']
+age_intervals = [6, 18, 32, 52, 72, 100]
+
+
+def age_to_group(age):
+    return float(next(x[0]+1 for x in enumerate(age_intervals) if x[1] >= age))
+
+
+def cross_validation_search(model, model_name, data, labels):
```

Comment:
Instead of printing the parameters and predictions, you should consider returning them in an object (see namedtuple for simple classless python objects), and returning these result objects to caller, where caller is going to decide what to do with them (print, store, do something else)